engine/flink/components-api/src/main/scala/pl/touk/nussknacker/engine/flink/api/process/FlinkSource.scala

```diff
@@ -45,16 +45,71 @@ trait FlinkSourceTestSupport[Raw] extends SourceTestSupport[Raw] { self: Source
   * @tparam Raw - type of raw event that is generated by flink source function.
   *             This is needed to handle e.g. syntax suggestions in UI (in sources with explicite @MethodToInvoke).
   */
-trait BasicFlinkSource[Raw] extends FlinkSource with FlinkIntermediateRawSource[Raw] {
+trait BasicFlinkSource[Raw]
+    extends FlinkSource
+    with CustomizableContextInitializerSource[Raw]
+    with CustomizableTimestampWatermarkHandlerSource[Raw]
+    with ExplicitTypeInformationSource[Raw] {
 
   @silent("deprecated")
   def flinkSourceFunction: SourceFunction[Raw]
 
+  override def contextInitializer: ContextInitializer[Raw] = new BasicContextInitializer[Raw](Unknown)
+
   override def sourceStream(
       env: StreamExecutionEnvironment,
       flinkNodeContext: FlinkCustomNodeContext
   ): DataStream[Context] = {
-    prepareSourceStream(env, flinkNodeContext, flinkSourceFunction)
+    FlinkIntermediateRawSourceUtils.prepareSourceStream(
+      env = env,
+      flinkNodeContext = flinkNodeContext,
+      sourceFunction = flinkSourceFunction,
+      typeInformation = typeInformation,
+      timestampWatermarkHandler = timestampAssigner,
+      contextInitializer = contextInitializer
+    )
+  }
+
+}
+
+/**
+ * Source providing default logic for transforming a `DataStreamSource[Raw]` into `DataStream[Context]`. Fulfills
+ * the same role as `BasicFlinkSource`, but is based on `DataStreamSource` instead of `SourceFunction`.
+ * 
+ * @tparam Raw - type of raw event that is generated by flink source function.
+ */
+trait StandardFlinkSource[Raw]
```

Comment:
"stadnard", "basic", "IntermediateRaw"? If you were a new developer that have to work on top of this change, it would be easy to pick the correct one? Let's remove some of them :)

Reply:
Right, I've removed Basic and IntermediateRaw.